### ISSUE TITLE
Crash when undeleting with a wildcard. 

### ISSUE THREAD

**Issue by rewolff** [neomutt/neomutt#2668]
## Expected Behaviour
I wanted to COPY a bunch of messages from a mailbox (zipped mbox format) to another mbox format file. So I tagged and saved a few messages as I found the ones I wanted. Those are now "marked for deletion" and I didn't want that So I typed shift-U, * to undelete everything. At that moment I got: Segmentation fault (core dumped)

## Actual Behaviour

crashed. 
If NeoMutt crashed, did your OS create a 'coredump' file?
yup. 

## Steps to Reproduce
see above. 

## How often does this happen?

- Always Yep!

## When did it start to happen?

- When I first had this task in neomutt. 

## NeoMutt Version

```
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.4.0-47-generic (x86_64)
ncurses: ncurses 6.2.20200212 (compiled with 6.2.20200212)
libidn: 1.33 (compiled with 1.33)

Configure options: 

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -DNCURSES_WIDECHAR -I/usr/include

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls -gpgme -gss -hcache -homespool +idn +inotify 
  -locales_hack -lua +meta -mixmaster +nls -notmuch -openssl +pgp +regex -sasl 
  +smime -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version Ubuntu 20.04
* Were you using multiple copies of NeoMutt at once? Yeah, but on different mailboxes, not the saved/archived one that causes this problem. I double checked with another mailbox: Same crash. 
* Were you using 'screen' or 'tmux'? no. 
* Is your email local (maildir) or remote (IMAP)? zipped mbox, unzipped mbox. (these two tested, both crash).

**Comment by rewolff** [comment]
Update: system says: segmentation fault (core dumped) but I can't find the core file. I trust such messages: In the past my computer didn't lie.

**Comment by flatcap** [comment]
Sorry about that, but...
Known bug, #2653, already fixed (in commit e25b8db).
There's a new release due on Friday.

**Comment by rewolff** [comment]
OK. Great. I can find a workaround. (The crash leaves the mailbox untouched; that's what I wanted. Alternatively I can mark the mailbox readonly and exit.). 
I'm running a reasonably recent git pull. A week too old for this fix. 
Pulled updated. Thanks for the help and "quick" (-10 days) fix. :-)
